### ISSUE TITLE
APK Download?

### ISSUE THREAD

**Issue by IzzySoft** [pyamsoft/fridgefriend#163]
Nice to see this FOSS app, and completely tracker-free at that – congrats! Would you mind making the APK available outside Google's "walled garden" – e.g. by attaching it to your [releases](https://github.com/pyamsoft/fridgefriend/releases) here at Github, or even bring it to [F-Droid](https://f-droid.org/) where the FOSS apps live for the privacy-aware users to get? Thanks a lot in advance!

**Comment by pyamsoft** [comment]
Hi, sure I can publish an APK and attach it as a GitHub release. I am fixing some bugs in the current version and dealing with the recent migration to Jetpack Compose, but should be able to cut a new release within the month.

Is there a specific name format you would prefer? For example release.apk, or fridgefriend-x.y.z.apk?

Please be aware though that sadly the app is not completely Google free. I use the in-app billing library which is tied to the proprietary Google Play Services framework so I don't believe this can be included on FDroid. So, to be fair, it's not completely FOSS and that Readme may need a slight update to clarify that.

**Comment by IzzySoft** [comment]
> should be able to cut a new release within the month.

That sounds good! If you let me know then, I gladly make your app available via [my F-Droid repo](https://apt.izzysoft.de/fdroid), for starters.

> Is there a specific name format you would prefer? For example release.apk, or fridgefriend-x.y.z.apk?

As long as it's called `*.apk`, it's fine (special naming conventions would only be needed should you provide builds of different flavors). My updater auto-renames APKs to `<packageName>_<versionCode>.apk` anyway. Just the tag name should match either `versionName` or `versionCode`, while prefixes or suffixes in the tag name are fine (e.g. tag name `v1.2.3` for `versionName: 1.2.3`) as my updater uses those to see if there's something it should fetch. So your current tagging matches that quite fine, as you're using `versionCode` as tag name.

> I use the in-app billing library which is tied to the proprietary Google Play Services framework so I don't believe this can be included on FDroid.

That's unfortunately correct – but still acceptable with my repo, which has slightly laxer inclusion criteria and thus allows for such "little slips". I sometimes call that "appetizers" for devs new to the "foss-and-privacy-world", to give them a start – and often this results in a completely free build flavor that goes to F-Droid.

With F-Droid, as well as with my repo, we usually prefer "donation links" – which are shown right next to the app description. We have special fields for special services like OpenCollective or Liberapay, but also for Bitcoin, Litecoin etc – and a generic "donation URL" for all that doesn't fit otherwise. And we encourage our users to use them.

Disclosure: saying "we" for one means "F-Droid's repo and mine" – but I'm also one of the F-Droid maintainers.

> So, to be fair, it's not completely FOSS and that Readme may need a slight update to clarify that.

Wow. Yes, true! And the "wow" is because that's the first time I heard that from an author without prompting them to. Amazing. Thanks a lot! May I ask which way you use the billing client? Is it just an easy payment method for donations – or does a payment promote the app to free some "pro features" otherwise hidden? Because if it's just for donations (without obligations), there might be replacements for that. Not that I have some of those in mind already (that section is still missing from [my F-Droid snippet](https://gitlab.com/-/snippets/1896503) which already lists replacements for push notifications, analytics and more), but I could check what my library definitions have to offer (and add a new section to that snippet).

**Comment by pyamsoft** [comment]
The in-app billing just serves as a way to receive compliant donation-like monetization on the Google Play store. All of my apps are free and open and fully featured, that will never change - there will never be an obligation to make an in-app purchase to unlock a feature. I write enough bugs as it is without charging money for them.

I've thought about a donation URL style in the past though ultimately don't have enough free time to maintain two separate builds for two different stores, so for the foreseeable short term, I'll be staying with the in-app billing client from Google.

I've used the IzzySoft fdroid repo in the past a ton and greatly appreciate your work, thanks for all you do to promote Foss on Android!

The next release will have the APK attached and I'll ping this issue when it's done - it may take one or two weeks as life gets in the way.

**Comment by IzzySoft** [comment]
It's ready when it's ready – take your time as you need. I much appreciate your openness! And thanks for your kind words, too!

If you want to, I can skim my library definitions for payment libs, to see if something fitting might be noted there. No pressure with that, though – just thinking if there's a lib fitting the bill for both sites, you might be interested. If not, I'm fine with keeping the app in my repo (once it's there) and bring it to F-Droid later (once a solution for that exists).

Looking forward to your word then, and the tag with the APK attached when it's ready :smiley:

**Comment by pyamsoft** [comment]
Hello,

I have just tagged and released version 38, along with an attached APK  named `app-release.apk`. I hope this will be sufficient for what you are needing. Please let me know if I can be of any help.

**Comment by IzzySoft** [comment]
Thanks a lot – and yes, fits well! Your app will [become available in my repo](https://apt.izzysoft.de/fdroid/index/apk/com.pyamsoft.fridge) with the next sync around 7 pm UTC today. I took screenshots and description from Play with some minor adjustments (description concerning the "fully FOSS", and removing frames from screenshots to make them better fit for small screens).

I tap my hat for your openness, but you're a bit harsh with yourself and your app: isn't it only the Play Billing library that's not FOSS? It almost sounds like other parts are affected, too.

I've checked my library definitions for possible alternatives. While most are either just other libs for Google's billing or something similar (Amazon, AliPay & Co), two candidates remain which you might wish to take a look at:

* [Android Donations Lib](https://github.com/SufficientlySecure/donations) (Apache-2.0)
* [stripe-android](https://github.com/stripe/stripe-android) (MIT)

The former would fit quite well (at least license-wise; I didn't check the dependencies).

----

As the original question is now answered, I guess this issue can be closed (unless there are questions left from your side). Once more, thanks a lot!
